### PR TITLE
custom types from config for multimedia module

### PR DESCRIPTION
**Pull request by didoda**
This PR provides the possibility to customize the types to handle in multimedia module, using a configuration.
For instance:
```
// default: 'application', 'audio', 'b_e_file', 'image', 'video'
$config['multimedia'] = array(
    'types' => array('audio', 'image', 'video'),
);
```

### CODE REVIEW

**Comment by stefanorosanelli** [comment]
This information should be read from API using something like:
```

File: bedita-app/controllers/modules/multimedia_controller.php [hunk 1]
```diff
@@ -20,85 +20,110 @@
...
+        parent::beditaBeforeFilter();
+        $objectTypes = Configure::read('multimedia.types');
+        if ($objectTypes) {
+            $this->object_types = $objectTypes;
+        }
+        foreach ($this->object_types as $type) {
+            $models[] = Inflector::camelize($type);
+        }
+        $mime = Configure::read('validate_resource.mime');
+        foreach ($mime as $modelMyme => $data) {
```
Comment (fquffio) [inline bedita-app/controllers/modules/multimedia_controller.php]: [minor] Typo: `modelMime`. 😄

**Review by fquffio — APPROVED** [review]
What if we access a disabled media type directly?

Imagine we have disabled video and object 123 is a video: `/multimedia/view/123` would result in an error?

**Comment by didoda** [comment]
Good point, @fquffio We should handle `view/:not_allowed_item_id` case

**Review by fquffio — APPROVED** [review]

File: bedita-app/controllers/modules/multimedia_controller.php [hunk 2]
```diff
@@ -20,85 +20,117 @@
...
+     * @var array
+     */
+    var $uses = array('BEObject', 'Application', 'Stream', 'Image', 'Audio', 'Video', 'Tree', 'User', 'Group', 'Category', 'BEFile');
+
+    /**
+     * The object types for module multimedia. Can be customized in configure 'multimedia.types'
+     * 
+     * @var array
+     */
+    protected $object_types = array('application', 'audio', 'b_e_file', 'image', 'video');
```
Comment (batopa) [inline bedita-app/controllers/modules/multimedia_controller.php]: should be `$objectTypes`

**Review by batopa — CHANGES_REQUESTED** [review]

**Review by batopa — APPROVED** [review]

**Pull request merged**